src/main/scala/rules/Consumer.scala

```diff
@@ -199,6 +236,44 @@ object consumer extends ConsumptionRules {
               Q(s2, h, Unit, v2)
             }))
 
+      case ite @ ast.CondExp(e0, a1, a2) if !a.isPure && Verifier.config.moreJoins() =>
+        val condExpRecord = new CondExpRecord(ite, s, v.decider.pcs, "consume")
+        val uidCondExp = SymbExLogger.currentLog().openScope(condExpRecord)
+
+        eval(s, e0, pve, v)((s1, t0, v1) =>
+          joiner.join[(Heap, Term), (Heap, Term)](s1, v1, resetState = false)((s1, v1, QB) => {
+            branch(s1, t0, v1)(
+              (s2, v2) => consumeR(s2, h, a1, pve, v2)((s3, h1, t1, v3) => {
+                SymbExLogger.currentLog().closeScope(uidCondExp)
+                QB(s3, (h1, t1), v3)
+              }),
+              (s2, v2) => consumeR(s2, h, a2, pve, v2)((s3, h1, t1, v3) => {
+                SymbExLogger.currentLog().closeScope(uidCondExp)
+                QB(s3, (h1, t1), v3)
+              }))
+          })(entries => {
+            val s2 = entries match {
+              case Seq(entry) => // One branch is dead
+                (entry.s, entry.data)
+              case Seq(entry1, entry2) => // Both branches are alive
+                val mergedData = (
+                  State.mergeHeap(
+                    entry1.data._1, And(entry1.pathConditions.branchConditions),
+                    entry2.data._1, And(entry2.pathConditions.branchConditions),
+                  ),
+                  // Asume that entry1.pcs is inverse of entry2.pcs
```

Comment:
Unclear: do you create an SMT assumption here, or is there some assumption underlying the next line of code? It is also not clear how sets of path conditions can be inverses of each other.

Typo in "asume"